src/mono/mono/metadata/marshal-lightweight.c

```diff
@@ -2798,19 +2832,33 @@ emit_managed_wrapper_ilgen (MonoMethodBuilder *mb, MonoMethodSignature *invoke_s
 	for (i = 0; i < sig->param_count; i ++) {
 		MonoType *t = sig->params [i];
 		MonoMarshalSpec *spec = mspecs [i + 1];
+		int csig_argnum = get_csig_argnum (i, m);
 
 		if (spec && spec->native == MONO_NATIVE_CUSTOM) {
-			tmp_locals [i] = mono_emit_marshal (m, i, t, mspecs [i + 1], 0,  &csig->params [i], MARSHAL_ACTION_MANAGED_CONV_IN);
+			tmp_locals [i] = mono_emit_marshal (m, csig_argnum, t, mspecs [i + 1], 0,  &csig->params [csig_argnum], MARSHAL_ACTION_MANAGED_CONV_IN);
 		} else {
 			switch (t->type) {
+			case MONO_TYPE_VALUETYPE:
```

Comment:
do you need a MONO_TYPE_GENERICINST case to handle `SwiftSelf<T>` or  other generics?

Reply:
I believe currently we do not plan to support `SwiftSelf<T>` in reverse pinvokes https://github.com/dotnet/runtime/pull/103576#issue-2357391606

cc: @kotlarmilos

Reply:
https://github.com/dotnet/runtime/pull/103576#issue-2357391606